src/dimwits/math.hpp

```diff
@@ -0,0 +1,113 @@
+namespace math {
+
+template< typename T >
+constexpr double e = 2.71828182845904523536028747135266249775724709369995;
+
+struct Exp{
```

Comment:
> I wish these had some sort of comment stating what they did.

The name choices `Exp`, `Log`, and `Pow` inform what structures do. Regarding the implementation, I'll point refer to the  [C++ Source Formatting and Style](https://github.com/njoy/DevelopersGuide/blob/master/Style/Code.md#5-documentation) guide. From entry 5

> Annotations aside, comments should not be used to describe what code does or how it works
> + prefer literate, self-documenting source code

- - -

> How do I know you aren't making up this algorithm? (I don't believe you are.)

I might argue all algorithms are made up :stuck_out_tongue: 

That said, these algorithms didn't come out of a textbook or anything; I sat down with some notebook paper for a few minutes. I've broken up the implementation to be a bit more explicit to address this concern, but there's really nothing exotic here and well within an engineering undergraduate curriculum.

+ `Exp`: repeated multiplication falling back to a Maclaurin series when the residual argument is near zero. 
+ `Log`: repeated division falling back to Newton-Raphson iteration when the residual argument is near unity, with an initial guess generated using the Mercator series.
+ `Pow`: application of logarithm properties 
---
> Do I need to become familiar with ways of implementing an exponential (or log) function to know what is going on? Sure, it would be helpful if I knew all that stuff, but it shouldn't be a pre-requisite to using this or to approving it.

For using it, of course not; encapsulation means a knowledge of the API will suffice. That said, for each class here there's a single method signature exposed in the interface

```c++
result = Exp::evaluate( arg );
result = Log::evaluate( arg );
result = Pow::evaluate( base, exponent );
```
If a user still felt uncertain about what `Pow` does, there'll be unit tests to reference for usage.

Regarding review and approval, an understanding of the implementation isn't necessary to begin but should certainly be an outcome! Part of the value proposition of code review is knowledge transfer and elimination of a single point of failure.